src/test/java/com/synopsys/integration/detect/ApplicationUpdaterTest.java

```diff
@@ -47,12 +46,14 @@ public class ApplicationUpdaterTest {
     
     @Test
     public void testCanSelfUpdate() {
-        Assertions.assertTrue(new ApplicationUpdater(utility, successArgs).canSelfUpdate());
+        Assertions.assertTrue(new ApplicationUpdater(new ApplicationUpdaterUtility(), successArgs).canSelfUpdate());
     }
     
     @Test
     public void testCanSelfUpdateWithoutBdUrl() {
-        Assertions.assertFalse(new ApplicationUpdater(utility, failureArgs).canSelfUpdate());
+        ApplicationUpdaterUtility mockedUtility = Mockito.mock(ApplicationUpdaterUtility.class);
+        Mockito.when(mockedUtility.getSysEnvProperty(ApplicationUpdater.SYS_ENV_PROP_BLACKDUCK_URL)).thenReturn(null);
+        Assertions.assertFalse(new ApplicationUpdater(mockedUtility, failureArgs).canSelfUpdate());
```

Comment:
This is the failing test on 8.9.z. The failure is due to the use of System environment property for Black Duck URL which is subverting the test case. By mocking the load of that System environment property, test case is able to proceed as expected.